lib/WebSocket.js

```diff
@@ -632,6 +634,16 @@ function initAsClient (address, protocols, options) {
 
   this._req = httpObj.get(requestOptions);
 
+  if (options.handshakeTimeout) {
+    this._req.setTimeout(options.handshakeTimeout, () => {
+      const err = new Error('opening handshake has timed out');
+      err.code = 'TIMEOUT';
```

Comment:
Do you need `code`? It's useful and many Node.js errors have this property but currently we don't have this property on our errors. I prefer to not add it for consistency and maybe revisit later.